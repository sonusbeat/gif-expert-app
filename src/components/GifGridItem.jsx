const GifGridItem = ({ title, url }) => {

  return (
    <figure className="card animate__animated animate__fadeIn">
      <img src={ url } alt={ title } />
      <figcaption className="card__title">{ title }</figcaption>
    </figure>
  );
}

export default GifGridItem;