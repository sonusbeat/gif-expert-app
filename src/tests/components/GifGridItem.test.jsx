import { shallow } from "enzyme";
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
  const title = "Dragon Ball";
  const url   = "https://localhost.com/dragon-ball.jpg";
  const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );
  const div = wrapper.find('div');
  const className = div.prop('className');

  test('Deberia coincidir con el Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("El pÃ¡rrafo debe coincidir con el valor de la constante title", () => {
    const paragraph = wrapper.find('p');

    expect(paragraph.text().trim()).toBe(title);
  });

  test("El atributo url de la imagen debe coincidir con la propiedad url", () => {
    const image = wrapper.find('img');

    expect(image.props().src).toBe(url);
  });

  test("El atributo alt de la imagen debe coincidir con la propiedad title", () => {
    const image = wrapper.find('img');

    expect(image.props().alt).toBe(title);
  });

  test('El div que envuelve la imagen debe contener la clase: "card"', () => {
    expect(className.includes('card')).toBe(true);
  });

  test('El div que envuelve la imagen debe contener las clases: "animate__animated" y "animate__fadeIn"', () => {
    expect(className.includes('animate__animated')).toBe(true);
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});