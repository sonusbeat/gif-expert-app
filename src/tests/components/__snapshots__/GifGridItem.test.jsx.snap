// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Deberia coincidir con el Snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Dragon Ball"
    src="https://localhost.com/dragon-ball.jpg"
  />
  <p>
    Dragon Ball
  </p>
</div>
`;
